src/subdataframe/subdataframe.jl

```diff
@@ -181,7 +181,16 @@ Base.@propagate_inbounds function Base.setindex!(sdf::SubDataFrame, val::Any, id
     setindex!(sdf, val, idx[1], idx[2])
 end
 Base.@propagate_inbounds function Base.setindex!(sdf::SubDataFrame, val::Any, ::Colon, colinds::Any)
-    parent(sdf)[rows(sdf), parentcols(index(sdf), colinds)] = val
+    if colinds isa SymbolOrString && getfield(sdf, :colindex) isa Index &&
+       val isa AbstractVector && columnindex(sdf, colinds) == 0 && nrow(sdf) == length(val)
```

Comment:
When `nrow(sdf) == length(val)` isn't true, an error will eventually be thrown by the other branch, right? Maybe it would be clearer to check this here. Taking a completely different path depending on the length makes me feel uneasy.

Also:
```suggestion
        val isa AbstractVector && columnindex(sdf, colinds) == 0 && nrow(sdf) == length(val)
```

Reply:
OK